stabilize normalization results by modifying sampling process

leahincom opened the pull request:
`normalize.py`
- random 하게 sampling 하여 결과가 일정하지 않은 문제 해결
- 샘플링한 데이터에 `autonormalize`를 수행해서 normalize 대상을 파악하는 과정 반복 (`default: 5`로 설정해둠)
- 위의 결과를 이용해 최종적으로 기존 데이터에 대해 `normalize` 수행

leahincom committed:
stabilize normalization results by modifying sampling process

cezanne commented:
기존 방식처럼 dataframe의 sample() 함수는 random하게 sampling을 잘 할겁니다. 굳이 재반복해서 sampling을 하여 교집합을 찾을 필요는 없어요.. 
다시 설명을 하면, sampling한 것으로 autonormalize 후 결과값이 나올 것이고, 이 과정을 수차례 반복한 결과값들의 공통 컬럼들을 기준으로 최종 normalize를 하는 것으로 하자는 것입니다.

leahincom commented:
> 기존 방식처럼 dataframe의 sample() 함수는 random하게 sampling을 잘 할겁니다. 굳이 재반복해서 sampling을 하여 교집합을 찾을 필요는 없어요..
> 다시 설명을 하면, sampling한 것으로 autonormalize 후 결과값이 나올 것이고, 이 과정을 수차례 반복한 결과값들의 공통 컬럼들을 기준으로 최종 normalize를 하는 것으로 하자는 것입니다.

넵! 참고해서 수정한 후 다시 PR 올리겠습니다.
감사합니다 :)

cezanne commented on featuretools/mkfeat/normalize.py:
```diff
@@ -1,33 +1,41 @@
+        for _ in range(5):  
+            df = df.sample(n=1000)
+            es = an.auto_entityset(df, index=key_colname, accuracy=0.98)
+            entities.update(es.entities[1:])
```
현재 코드는 원하는 방식은 아닌 거 같습니다. 찾아낸 entity와 relationship들의 교집합이 되어야 될 거 같습니다. 그런데 단순히 set을 써서 합집합, 교집합을 얻어내는 것이라기 보다는 entity의 컬럼의 교집합이 되어야 하지 않을지? 설명이 어렵네요. ㅠ

cezanne committed:
Fix FeatureImportance for separate label file case

leahincom committed:
add argument conditions to transform operator: GreaterThanPrevious

leahincom committed:
add argument conditions to transform operator: LessThanPrevious

leahincom committed:
add argument conditions to aggregation operator: CountInsideRange

leahincom committed:
add argument conditions to aggregation operator: CountOutsideRange

cezanne committed:
Handle the case a key column has a non-unique

leahincom committed:
add argument conditions and refactor aggregation operator: MaxConsecutiveFalse

leahincom committed:
add argument conditions and refactor aggregation operator: MaxConsecutiveNegatives

leahincom committed:
add argument conditions and refactor aggregation operator: MaxConsecutivePositives

leahincom committed:
add argument conditions and refactor aggregation operator: MaxConsecutiveTrue

leahincom committed:
add argument conditions and refactor aggregation operator: MaxConsecutiveZeros

cezanne committed:
Add support for auto key

Now, column spec will be OK if it has no key column. In that case,
a temporary key is generated, whose column name is 'id' or 'id_xxx'.

cezanne committed:
Manage the bug of autonormalize for multi key

cezanne committed:
Fix error for non-numeric columns of label file

A label file for importance should have exactly one label column.
Old implementation assumed that a label file has only 1 numeric column.
Also, apply validation for FeatureImportance.

cezanne committed:
Add ERR_DATA_LABEL_COUNT_MISMATCH

leahincom committed:
Merge remote-tracking branch 'fork/dev' into dev
